src/web/entity/navigation.ts

```diff
@@ -5,19 +5,44 @@
 
 import {isDefined, isFunction} from 'gmp/utils/identity';
 
+type NavigateFunc = (path: string) => void;
+
+interface NavigateObj {
+  navigate: NavigateFunc;
+}
+
+type Navigate = NavigateFunc | NavigateObj;
+
+interface GotoDetailsObj {
+  data: {
+    id: string;
+  };
+}
+
+interface GotoDetailsFunc {
```

Comment:
Suggestion: since there is only definition, we could use `type`
```suggestion
type GotoDetailsFunc = ({data}: GotoDetailsObj) => void;

type GotoListFunc = () => void;

```